### PR TITLE
Support attachable aws roles

### PR DESCRIPTION
**Pull request by michieldewilde**
## Type of change

- [x] New feature (non-breaking change that adds functionality)

### Development

- [x] Lint rules pass locally
- [x] The code changed/added as part of this pull request has been covered with tests
- [x] All tests related to the changed code pass in development

### Code review

- [x] This pull request has a descriptive title and information useful to a reviewer. There may be a screenshot or screencast attached
- [x] Pull Request is no longer marked as "draft"
- [ ] Reviewers have been assigned
- [x] Changes have been reviewed by at least one other engineer
- [x] The target branch is `future` unless the change is going directly into production

### CODE REVIEW

File: spacelift/resource_aws_integration.go [hunk 1]
```diff
@@ -0,0 +1,176 @@
...
+				Default:     900,
+				Optional:    true,
+			},
+			"labels": {
+				Type:        schema.TypeSet,
+				Description: "Labels to set on the integration",
+				Elem:        &schema.Schema{Type: schema.TypeString},
+				Optional:    true,
+			},
+			"legacy": {
```
Comment (adamconnelly) [inline spacelift/resource_aws_integration.go:71]: Thinking about this more, maybe we shouldn't include `legacy` on the resource or data source. I can't think of a scenario where the info would be useful within Terraform, and it's an extra thing to deprecate later on.

File: spacelift/resource_aws_integration_attachment.go [hunk 2]
```diff
@@ -0,0 +1,175 @@
...
+				Optional:     true,
+				ForceNew:     true,
+			},
+			"read": {
+				Type:        schema.TypeBool,
+				Description: "Indicates whether this attachment is used for read operations",
+				Optional:    true,
+				Default:     true,
+			},
+			"subscription_id": {
```
Comment (adamconnelly) [inline spacelift/resource_aws_integration_attachment.go:58]: You don't need this for AWS.

File: spacelift/resource_aws_integration_attachment.go [hunk 3]
```diff
@@ -0,0 +1,175 @@
...
+
+	variables := map[string]interface{}{
+		"id":    toID(d.Get("attachment_id")),
+		"read":  graphql.Boolean(d.Get("read").(bool)),
+		"write": graphql.Boolean(d.Get("write").(bool)),
+	}
+
+	if subscriptionID, ok := d.GetOk("subscription_id"); ok {
+		variables["subscriptionId"] = toOptionalString(subscriptionID)
+	}
```
Comment (adamconnelly) [inline spacelift/resource_aws_integration_attachment.go:152]: This can go.

File: spacelift/data_aws_integration.go [hunk 4]
```diff
@@ -0,0 +1,91 @@
...
+)
+
+func dataAWSIntegration() *schema.Resource {
+	return &schema.Resource{
+		Description: "" +
+			"`spacelift_aws_integration` represents an integration with an AWS " +
+			"account. This integration is account-level and needs to be explicitly " +
+			"attached to individual stacks in order to take effect." +
+			"\n\n" +
+			"Note: when assuming credentials for **shared worker**, Spacelift will use `$accountName-$integrationID@$stackID-suffix` " +
```
Comment (adamconnelly) [inline spacelift/data_aws_integration.go:21]:
```suggestion
			"Note: when assuming credentials for **shared workers**, Spacelift will use `$accountName-$integrationID@$stackID-$suffix` " +
```

File: spacelift/data_aws_integration.go [hunk 5]
```diff
@@ -0,0 +1,91 @@
...
+			"or `$accountName-$integrationID@$moduleID-suffix` as [external ID](https://docs.aws.amazon.com/IAM/latest/UserGuide/id_roles_create_for-user_externalid.html) " +
+			"and `$runID@$stackID@$accountName` truncated to 64 characters as [session ID](https://docs.aws.amazon.com/STS/latest/APIReference/API_AssumeRole)," +
+			"Suffix will be read or write.",
```
Comment (adamconnelly) [inline spacelift/data_aws_integration.go:24]:
```suggestion
			"`$suffix` will be `read` or `write`.",
```

File: spacelift/resource_aws_integration.go [hunk 6]
```diff
@@ -0,0 +1,176 @@
...
+)
+
+func resourceAWSIntegration() *schema.Resource {
+	return &schema.Resource{
+		Description: "" +
+			"`spacelift_aws_integration` represents an integration with an AWS " +
+			"account. This integration is account-level and needs to be explicitly " +
+			"attached to individual stacks in order to take effect." +
+			"\n\n" +
+			"Note: when assuming credentials for **shared worker**, Spacelift will use `$accountName-$integrationID@$stackID-suffix` " +
```
Comment (adamconnelly) [inline spacelift/resource_aws_integration.go:21]:
```suggestion
			"Note: when assuming credentials for **shared workers**, Spacelift will use `$accountName-$integrationID@$stackID-$suffix` " +
```

File: spacelift/resource_aws_integration.go [hunk 7]
```diff
@@ -0,0 +1,176 @@
...
+			"or `$accountName-$integrationID@$moduleID-suffix` as [external ID](https://docs.aws.amazon.com/IAM/latest/UserGuide/id_roles_create_for-user_externalid.html) " +
```
Comment (adamconnelly) [inline spacelift/resource_aws_integration.go:22]:
```suggestion
			"or `$accountName-$integrationID@$moduleID-$suffix` as [external ID](https://docs.aws.amazon.com/IAM/latest/UserGuide/id_roles_create_for-user_externalid.html) " +
```

File: spacelift/resource_aws_integration.go [hunk 8]
```diff
@@ -0,0 +1,176 @@
...
+			"and `$runID@$stackID@$accountName` truncated to 64 characters as [session ID](https://docs.aws.amazon.com/STS/latest/APIReference/API_AssumeRole)," +
+			"Suffix will be read or write.",
```
Comment (adamconnelly) [inline spacelift/resource_aws_integration.go:24]:
```suggestion
			"`$suffix` will be `read` or `write`.",
```

File: spacelift/resource_aws_integration.go (hunk 7 above)
Comment (michieldewilde) [reply spacelift/resource_aws_integration.go]: fixed

**Pull request merged**